internal/venice-common/src/main/java/com/linkedin/venice/pubsub/PubSubPositionFactory.java

```diff
@@ -0,0 +1,58 @@
+package com.linkedin.venice.pubsub;
+
+import com.linkedin.venice.exceptions.VeniceException;
+import com.linkedin.venice.pubsub.adapter.kafka.ApacheKafkaOffsetPosition;
+import com.linkedin.venice.pubsub.api.PubSubPosition;
+import com.linkedin.venice.pubsub.api.PubSubPositionWireFormat;
+import com.linkedin.venice.serialization.avro.AvroProtocolDefinition;
+import com.linkedin.venice.serialization.avro.InternalAvroSpecificSerializer;
+import java.io.IOException;
+
+
+/**
+ * Factory class for creating PubSubPosition objects from wire format
+ */
+public class PubSubPositionFactory {
+  public static final PubSubPositionFactory INSTANCE = new PubSubPositionFactory();
+
+  /**
+   * Converts a wire format position to a PubSubPosition
+   * @param positionWireFormat the wire format position
+   * @return concrete position object represented by the wire format
+   */
+  PubSubPosition convertToPosition(PubSubPositionWireFormat positionWireFormat) {
+    if (positionWireFormat == null) {
+      throw new IllegalArgumentException("Cannot deserialize null wire format position");
+    }
+
+    switch (positionWireFormat.type) {
+      case PubSubPositionType.APACHE_KAFKA_OFFSET:
```

Comment:
Maybe worth noting with either javadoc or code comment, but the implication here is that all subscribers to a topic will need to be at some X version of this object before it can be converted.  Otherwise conversion will trigger exceptions.  When thinking about Venice core components, maybe not a bit deal, but since this will go into clients, it implies we'll need to be able to track those clients and get them to upgrade.

Reply:
Hmm, but what is the alternative? What could we do besides throwing if non-upgraded code encounters a type it cannot deal with?

Reply:
Been meaning to have as an inday project to have our jars signed with the release number and then emit that jar number to metrics so we can track this stuff better.  Not relevant to this rb, but something to consider if we're worried about tracking this stuff as it is deployed.

Reply:
@ZacAttack - I'm not sure if I fully understood your point. Here is what I think:
1. To begin with `PubSubPosition` should only be used by brand-new clients such as CC.
  As all clients will have the type of offset from the beginning, there won't be any issue

2. For the existing clients we need to support (initially) `long` to `PubSubPosition`(`ApacheKafkaOffsetPosition`)  conversion and vice versa.  This is not supported by the code in this PR. We'll create it in a separate PR.  Till then we will continue using `long`. 
 If we do not use `PubSubPosition` in such cases shouldn't be any issue.

Reply:
I don't have an alternative, I think this is the right thing to do.  I was just noting this property and suggesting a comment somewhere.

Reply:
Ack.  The comment you suggested should be added to the `PubSubPositionFactory` extension IMO. Let's tackle it later. Thanks!